src/lib/rule2G_test.ts

```diff
@@ -0,0 +1,71 @@
+import {html, render} from 'lit-html';
+import {rule2G} from './rule2G';
+
+describe('The function for rule 2G', () => {
+  let container: HTMLElement;
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  it('returns text content text nodes', () => {
+    render(
+      html`
+        <div id="foo">
+          Hello world
+        </div>
+      `,
+      container
+    );
+    const node = document.getElementById('foo')?.childNodes[0];
+    expect(rule2G(node!)).toBe('Hello world');
+  });
+
+  it('returns text content of text nodes as a flat string', () => {
+    render(
+      html`
+        <div id="foo">
+          Hello world newline
+        </div>
+      `,
+      container
+    );
+    const node = document.getElementById('foo')?.childNodes[0];
+    expect(rule2G(node!)).toBe('Hello world newline');
+  });
+
+  it('trims any excess whitespace from the resulting string', () => {
+    render(
+      html`
+        <div id="foo">
+          Hello world
+        </div>
+      `,
+      container
+    );
+    const node = document.getElementById('foo')?.childNodes[0];
+    expect(rule2G(node!)).toBe('Hello world');
+  });
```

Comment:
These 3 tests seem to all be testing the same thing to me (if one fails I think they'll all fail). Can you rewrite them to be testing only what they want to test. i.e. the first test should be `<div id="foo">Hello world</div>`, the second test should have different whitespace characters between the words, and the third test should have different whitespace characters at the beginning and end.

Reply:
Will do! I think that GTS may have changed this as originally I had newlines and trailing spaces.

Reply:
+1, I think you can just use  `document.createTextNode('...');` for these tests & that should allow you to test things more precisely.